public/global.js

```diff
@@ -215,5 +215,16 @@ function populateTextSectionContent() {
     $("#fillVariablesClose").click(onFillVarsClose);
     $("#fillVarsModal").on("hidden.bs.modal", onFillVarsClose);
     $("#fillVarsModal").on("shown.bs.modal", onFillVarsOpen);
+
+    $(window).bind('beforeunload', function(){
```

Comment:
> Does this bind a new handler each time it is called? Or is it just called once?

It is called a bit after the initial load of the page, which triggered this issue. The next commit moves this to be bound only when an input action happens.

- Just when doing little action like erasing few letters, it wont be bound, I suggest saving previous value of CodeMirror editor on a global variable and check if it is actually changed ( no pre-made CodeMirror event does that ).

Reply:
This does not answer my question. If the `inputRead` is fired multiple times, then there will be multiple event handlers for the same event. It is still buggy. When I delete characters using backspace, it doesn't trigger. When I press enters, I see this error in the console: "onloadwff.js:71 Assertion failed: Input argument is not an HTMLInputElement".
It is only triggered when I type some characters in the editor.
Why fiddle with the event binding while you can simply check for changes and return false in the handler instead of true if there are no changes? You can then listen on the `change` or `keydown` event for the editor and set that variable to true.

Reply:
> This does not answer my question. If the `inputRead` is fired multiple times, then there will be multiple event handlers for the same event. It is still buggy. When I delete characters using backspace, it doesn't trigger. When I press enters, I see this error in the console: "onloadwff.js:71 Assertion failed: Input argument is not an HTMLInputElement". It is only triggered when I type some characters in the editor. Why fiddle with the event binding while you can simply check for changes and return false in the handler instead of true if there are no changes? You can then listen on the `change` or `keydown` event for the editor and set that variable to true.

Thank you for letting me think that way, Yes, I totally agree with this. 

To detect changes in the editor, updating the template is managed between two files by default, so I set a global variable to detect such changes. I tested this with creating & updating the template, tell me if you find any issues.

Reply:
@fayez-nazzal I still can't see the code changes as explained in my earlier comment. I think it is a bad idea to bind on changes and unbind on response success. Simply bind on page load and return `false` from the unload event if any changes have been done (`changed==true`). No need to unbind at all because the page is dismissed. Just set the `changed` variable to `false` on the response success.

Reply:
@hammady  The `unbind` calls are removed now, I used the approach you suggest.

Just note that the `beforeunload` event in specific seemed different than other events, returning false will also trigger it ( At least on some browsers ), to cancel the event I had to return `undefined` instead.

Reply:
That's fine, I am not a front-end expert, good that you tested it on multiple browsers.